### PR TITLE
Allow user to set their nickname

### PR DESCRIPTION
**Pull request by anna-giang**
Upon login, check whether user has set their nickname, and redirect user to the `/createUserProfile.html` containing the form to set it up if they haven't yet. Store their nickname in datastore in an UserInfo entity.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 1]
```diff
@@ -35,13 +35,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     UserService userService = UserServiceFactory.getUserService();
     HashMap<String, String> userData = new HashMap<String, String>();
 
-    if (userService.isUserLoggedIn()) {
+    if (userService.isUserLoggedIn()) {
```
Comment (thanhdng) [inline portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java]: nit: remove trailing whitespace

File: portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+public class NicknameServlet extends HttpServlet {
+  
+  /** Checks whether user has set a nickname previously. Directs user to /createUserProfile.html to
+   * set their nickname, otherwise redirects back to /index.html */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (!userService.isUserLoggedIn()) { // If user not logged in, redirect them to login before coming back
+      String loginUrl = userService.createLoginURL("/nickname");
```
Comment (thanhdng) [inline portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java:43]: question: should this be a different URL?

**Review by thanhdng — COMMENTED** [review]
Some comments/questions

File: portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java (hunk 2 above)
Comment (anna-giang) [reply portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java:43]: The user should be redirected back there to set up their nickname at the "/nickname" URL after they login, which is why the redirect URL inputted there is also "/nickname".

**Review by thanhdng — APPROVED** [review]
LGTM, please fix the nit as well

File: portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java (hunk 1 above)
Comment (anna-giang) [reply portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java]: yep done!

**Pull request merged**